### ISSUE TITLE
effort_controllers/joint_position_controller from ROS1

### ISSUE THREAD

**Issue by tonylitianyu** [ros-controls/ros2_controllers#373]
ros1_controllers has **effort_controllers/joint_position_controller** that accepts position commands and targets an effort-controlled joint. But in this repo, the effort_controllers only accept group effort commands. Is it moved somewhere else or the feature is still missing?

**Comment by tonylitianyu** [comment]
To be more specific, I am trying to control an arm to a set joint position, but with limited torque, so that when an external force (larger than the torque limit) is present, the arm will deviate from the position (with some kind of compliant behavior). Once the external force is gone or become smaller, the arm will tend to move back to the set joint position. Basically some kind of passive motion. Back in ROS1, I can achieve such behavior with effort_controllers/joint_position_controller. I am wondering if ros2_controller has something similar to this.

**Comment by bmagyar** [comment]
This functionality is currently missing but it's fairly straight forward to
port it from ROS1. I'd happily review a PR for this

**Comment by tonylitianyu** [comment]
@bmagyar Just realize https://github.com/ros-controls/ros2_controllers/pull/198 has the group version for this a year ago, is there a reason why it has not merged yet? Would be happy to contribute to this.

**Comment by bmagyar** [comment]
It needs revival :)

**Comment by destogl** [comment]
@tonylitianyu you wanted to contribute to this. Is this still the case?

**Comment by Aadiv1999** [comment]
@bmagyar i am trying to build the effort_controller: joint_position_controller from [here](https://github.com/ros-controls/ros2_controllers/tree/effort_group_position_controller/effort_controllers), but I keep getting build errors for Humble, is there any new branch with a fix for this?

**Comment by bmagyar** [comment]
There was a sync a few days ago, if you `rosdep update && rosdep install --from-path src --ignore-src`, it should build fine again.

**Comment by Aadiv1999** [comment]
@bmagyar I tried that, but i still seem to be getting errors while building. To give you an idea of my setup:

- Ubuntu 22.04 
- ROS2 Humble
- In my home directory, ros2 workspace (ros2_ws)
- I have the binary install of ros2_controls and ros2_controllers (ros-humble-ros2-control and ros-humble-ros2-controllers)
- In ~/ros2_ws/src i have cloned the effort_controllers package from [here](https://github.com/ros-controls/ros2_controllers/tree/effort_group_position_controller/effort_controllers)

[here](https://drive.google.com/file/d/1OT8YuICxlYe7KYcne9Gp1ZbFrBrXBSrg/view?usp=sharing) is the error message I get while building.

**Comment by bmagyar** [comment]
Have you tried with --allow-overriding as per the error message? We
generally do not recommend having the same package you are trying to build
from source installed as binary too as ros2 seems to still have issues with
it.

**Comment by Aadiv1999** [comment]
@bmagyar yes I did try, it is still giving me build issues/issues in the code itself as shown in the attached file

**Comment by pky404** [comment]
@bmagyar @Karsten1987 I was able to build [_this_](https://github.com/ros-controls/ros2_controllers/tree/effort_group_position_controller) in my workspace(removed binary installation) but getting errors while running the Effort_Controllers/JointGroupPositionController although Effort_Controllers/JointGroupPositionController runs without error. 

My system:
- Ubuntu 20.04
- ROS2 Foxy

Here is the error : 
![effort_error](https://user-images.githubusercontent.com/75672764/210063138-9852240c-3312-4f83-bd41-7b3e5d47d96b.jpg)

**Comment by christophfroehlich** [comment]
Not needed any more as we have the PidController now.
